### PR TITLE
MM-55440: server-side feature usage

### PR DESCRIPTION
**Pull request by ifoukarakis**
#### Summary

- [x] Consider server-side events for feature usage. 
- [x] Use feature information available in event payload.
- [x] Adds events related to scheduled post .
- [x] Adds events related to read-only channels.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-55440

### CODE REVIEW

File: transform/mattermost-analytics/models/intermediate/product/features/mattermost/int_feature_daily_usage_per_user.sql [hunk 1]
```diff
@@ -1,35 +1,57 @@
...
+          prefix='count_',
+          quote_identifiers=False
+      ) }}
+    , {% for val in known_features %}
+         {% if not loop.first %} + {% endif -%} count_{{val}}
+    {%- endfor %} as count_known_features
     , count(event_id) as count_total
 from
-    {{ ref('int_feature_attribution') }}
+    {{ ref('int_mattermost_feature_attribution') }}
```
Comment (catalintomai) [inline transform/mattermost-analytics/models/intermediate/product/features/mattermost/int_feature_daily_usage_per_user.sql:47]: do we still need to keep `int_feature_attribution`?

**Review by catalintomai — APPROVED** [review]

File: transform/mattermost-analytics/models/intermediate/product/features/mattermost/int_feature_daily_usage_per_user.sql [hunk 2]
```diff
@@ -1,35 +1,57 @@
--- Forcing dependency due to load_feature_mappings macro
--- depends_on: {{ ref('tracking_plan') }}
-
 {{
     config({
-        "materialized": "table",
+        "materialized": "incremental",
```
Comment (catalintomai) [inline transform/mattermost-analytics/models/intermediate/product/features/mattermost/int_feature_daily_usage_per_user.sql:3]: @ifoukarakis , forgot to mention - are there any new client event names for the new features  to be added to the `tracking_plan`?

Comment (ifoukarakis) [reply transform/mattermost-analytics/models/intermediate/product/features/mattermost/int_feature_daily_usage_per_user.sql:3]: No. All new events are from server side. This includes read only channels and scheduled posts. features A few events are also available for guest accounts, which also exist on client-side. 

I'll create update confluence, documenting the known values used in feature names.

Comment (catalintomai) [reply transform/mattermost-analytics/models/intermediate/product/features/mattermost/int_feature_daily_usage_per_user.sql:3]: Thanks, documenting the feature events would be really helpful. I'm going through the list in https://docs.google.com/document/d/1u7UcVv6zaFIsR4YzhlJptZZSHv0M6ERqv9jPSGtp8po/edit?tab=t.0 - to identify the events that are not yet captured in code, so having a reference list (other than scouring PRs, tracking list or querying the RAW data) would help.

**Pull request merged**